Sources/PrimerSDK/Classes/User Interface/Root/PrimerRootViewController.swift

```diff
@@ -532,15 +490,36 @@ extension PrimerRootViewController: UIGestureRecognizerDelegate {
     }
 }
 
+
+extension PrimerRootViewController {
+    
+    private func handleErrorBasedOnSDKSettings(_ error: Error) {
+        
+        let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
+        if settings.isManualPaymentHandlingEnabled {
+            PrimerDelegateProxy.onResumeError(error)
+            handle(error: error)
+        } else {
+            Primer.shared.delegate?.primerDidFailWithError?(error, data: nil, decisionHandler: { errorDecision in
+                if let errorMessage = errorDecision?.additionalInfo?[.message] as? String {
+                    let merchantError = PrimerError.merchantError(message: errorMessage)
+                    self.handle(error: merchantError)
+                } else {
+                    self.handle(error: error)
```

Comment:
Remove this line, pass an error with an empty string as description.

Reply:
Addressed ✅ 
Lets see whether we could give better error code
```
static var emptyDescriptionError: NSError {
    NSError(domain: "", code: 0001)
}
```